flake.nix

```diff
@@ -35,6 +37,37 @@
             ./nix/hetzner.nix
           ];
         };
+
+        # working version on AWS
+        nixosConfigurations.jl4-aws-2505 = inputs.nixpkgs.lib.nixosSystem {
+          system = "x86_64-linux";
+
+          modules = [
+            (let stateVer = "25.05"; in {
+              system.stateVersion = stateVer;
+              # do not change the stateVersion even if you change nixpkgs.url above.
+              # the system is not fully stateless because the jl4-websessions component persists l4 programs to sqlite
```

Comment:
```suggestion
``` 
nit: the explanation is misleading - the system is in general not entirely stateless - that's not the important bit though - the issue is that some state is not compatible across `stateVersion`s. See e.g. https://github.com/NixOS/nixpkgs/blob/4b1164c3215f018c4442463a27689d973cffd750/nixos/modules/services/databases/postgresql.nix#L126

Reply:
right, this should keep stateVersions pinned for different deployments … the WT deployment will keep 24.11 and the legalese deployment will have 2505